dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/util/MyTaskProgressViewSkin.java

```diff
@@ -0,0 +1,162 @@
+
+        @Override
+        public void updateIndex(final int index) {
+            super.updateIndex(index);
+
+            /*
+             * I have no idea why this is necessary but it won't work without
+             * it. Shouldn't the updateItem method be enough?
+             */
+            if (index == -1) {
+                setGraphic(null);
+                getStyleClass().setAll("task-list-cell-empty");
+            }
+        }
+
+        @Override
+        protected void updateItem(final T task, final boolean empty) {
+            super.updateItem(task, empty);
+
+            this.task = task;
+
+            if (empty || task == null) {
+                getStyleClass().setAll("task-list-cell-empty");
+                setGraphic(null);
+            } else if (task != null) {
+                getStyleClass().setAll("task-list-cell");
+                progressBar.progressProperty().bind(task.progressProperty());
+                titleText.textProperty().bind(task.titleProperty());
+                messageText.textProperty().bind(task.messageProperty());
+                cancelButton.disableProperty().bind(
+                        Bindings.not(task.runningProperty()));
+
+                final Callback<T, Node> factory = getSkinnable().getGraphicFactory();
+                if (factory != null) {
+                    final Node graphic = factory.call(task);
+                    if (graphic != null) {
+                        BorderPane.setAlignment(graphic, Pos.CENTER);
+                        BorderPane.setMargin(graphic, new Insets(0, 4, 0, 0));
+                        borderPane.setLeft(graphic);
+                    }
+                } else {
+                	/*
+                	 * Really needed. The application might have used a graphic
+                	 * factory before and then disabled it. In this case the border
+                	 * pane might still have an old graphic in the left position.
+                	 */
+                    borderPane.setLeft(null);
+                }
+
+                setGraphic(borderPane);
+            }
+        }
+    }
+
+    private void popupCancelTask(final T task, final ActionEvent evt) {
+        final Button exitButton = (Button) CLOSECONFIRMATIONALERT.getDialogPane().lookupButton(
+                ButtonType.OK
+        );
+        final Button cancelButton = (Button) CLOSECONFIRMATIONALERT.getDialogPane().lookupButton(
+                ButtonType.CANCEL
+        );
+        exitButton.setText("Yes");
+        cancelButton.setText("No! I don't");
+        CLOSECONFIRMATIONALERT.setHeaderText("Are you really want to cancel this job");
+        CLOSECONFIRMATIONALERT.setContentText("Job will be cancelled. You can not recover this in future.");
+        final Optional<ButtonType> closeResponse = CLOSECONFIRMATIONALERT.showAndWait();
+        if (closeResponse.get().equals(ButtonType.OK)) {
+            if (task != null) {
+                task.cancel();
+            }
+        }else
```

Comment:
This else block should have curly braces to conform with our coding standards: [curly braces](https://github.com/SpectraLogic/spectralogic.github.com/wiki/Java-Style-Guide#conditional-statements)